lib/async/enumerable.rb

```diff
@@ -61,26 +81,25 @@ def async(max_fibers: nil)
           else
             send(source_method)
           end
-          fiber_limit = max_fibers || self.class.default_max_fibers
+          ::Async::Enumerator.new(source, **kwargs)
         else
           # If no source is defined, assume self is enumerable
-          source = self
-          fiber_limit = max_fibers
+          ::Async::Enumerator.new(self, **kwargs)
         end
-        ::Async::Enumerator.new(source, max_fibers: fiber_limit)
       end
     end
 
     module ClassMethods
       # Defines enumerable source for async operations.
       # @param method_name [Symbol] Method/ivar returning enumerable
-      # @param max_fibers [Integer, nil] Default concurrency limit
+      # @param max_fibers [Integer, nil] Default concurrency limit (kept for compatibility)
       def def_enumerator(method_name, max_fibers: nil)
         @enumerable_source = method_name
-        @default_max_fibers = max_fibers
+        # max_fibers parameter kept for backward compatibility but not used
+        # Users should pass max_fibers to .async() method instead
```

Comment:
i'd like to retain the ability to set `max_fibers` (and all the config options) through `def_enumerator`. we'd need to target/update the class-level config wherever we're included -- do we have a class level config? isn't that what `__async_enumerable_config` refers to?